cli/pcluster/cli_commands/compute_fleet_status_manager.py

```diff
@@ -0,0 +1,141 @@
+            compute_fleet_status = self._table.get_item(ConsistentRead=True, Key={"Id": self.COMPUTE_FLEET_STATUS_KEY})
+            if not compute_fleet_status or "Item" not in compute_fleet_status:
+                raise Exception("COMPUTE_FLEET status not found in db table")
+            return ComputeFleetStatus(compute_fleet_status["Item"][self.COMPUTE_FLEET_STATUS_ATTRIBUTE])
+        except Exception as e:
+            LOGGER.error("Failed when retrieving fleet status from DynamoDB with error %s", e)
+            return fallback
+
+    def update_status(self, current_status, next_status):
+        """Set compute fleet status."""
+        try:
+            self._table.put_item(
+                Item={"Id": self.COMPUTE_FLEET_STATUS_KEY, self.COMPUTE_FLEET_STATUS_ATTRIBUTE: str(next_status)},
+                ConditionExpression=Attr(self.COMPUTE_FLEET_STATUS_ATTRIBUTE).eq(str(current_status)),
+            )
+        except self._ddb_resource.meta.client.exceptions.ConditionalCheckFailedException as e:
+            raise ComputeFleetStatusManager.ConditionalStatusUpdateFailed(e)
+
+    def update_status_and_wait_transition(self, request_status, in_progress_status, final_status):
+        """
+        Update the status of the compute fleet and wait for a status transition.
+
+        It updates the status of the fleet to request_status and then waits for it to be updated to final_status,
+        by eventually transitioning through in_progress_status
+        """
+        compute_fleet_status = self.get_status()
+        if not compute_fleet_status:
+            raise Exception("Could not retrieve compute fleet status.")
+
+        if compute_fleet_status == final_status:
+            LOGGER.info("Compute fleet already in %s status.", final_status)
+            return
+
+        if compute_fleet_status not in {request_status, in_progress_status, final_status}:
+            self.update_status(current_status=compute_fleet_status, next_status=request_status)
+        LOGGER.info("Submitted compute fleet status transition request. Waiting for status update to start...")
+
+        compute_fleet_status = self._wait_for_status_transition(wait_on_status=request_status, timeout=180)
+        if compute_fleet_status == in_progress_status:
+            LOGGER.info(
+                "Compute fleet status transition is in progress. This operation might take a while to complete..."
+            )
+            compute_fleet_status = self._wait_for_status_transition(wait_on_status=in_progress_status, timeout=600)
+
+        if compute_fleet_status != final_status:
+            raise Exception(
+                "Unexpected final state {} probably due to a concurrent status update request.".format(
+                    compute_fleet_status
+                )
+            )
+        else:
+            LOGGER.info("Compute fleet status updated successfully.")
+
+    def _wait_for_status_transition(self, wait_on_status, timeout=300, retry_every_seconds=15):
```

Comment:
With this new method we are changing the behaviour of the start/stop commands.
I mean in past the commands were executed and returned without waiting for the transition.

Now we're waiting and checking for the status and the command could hang for a bit. I'd add a least a message before the sleep to notify the user that the command is waiting to the status change.

Reply:
If you look at `update_status_and_wait_transition` there are some messages shown before calling the wait function, in particular:
- Submitted compute fleet status transition request. Waiting for status update to start...
- Compute fleet status transition is in progress. This operation might take a while to complete...

Shall I reword them?

Reply:
Probably we should give some information to the user for each loop or at every minute.